tzkt/fetchTransactionTree.ts

```diff
@@ -46,85 +52,98 @@ async function fetchTransactions(
   return results;
 }
 
+function isUserWallet(wallet: Wallet) {
+  const address = wallet.address;
+  const isUserWallet = !(
+    wallet.address.startsWith("KT1") ||
+    exchangeAliases.some((alias) => alias === wallet.alias)
+  );
+  console.log(
+    "is Exchange wallet",
+    exchangeAliases.some((alias) => alias === wallet.alias)
+  );
+
+  console.log(`Wallet belongs to a user ${isUserWallet}`, wallet);
+  return isUserWallet;
+}
+
 function sleep(ms) {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
 
-async function fetchUniqueAddresses(
-  addresses: string[],
+async function fetchUniqueWallet(
+  wallets: Wallet[],
   start: Date,
   end: Date,
   direction: Direction.IN | Direction.OUT
-) {
+): Promise<Wallet[]> {
   const transactions = await fetchTransactions(
-    addresses,
+    wallets,
     start,
     end,
     direction === Direction.IN ? "target" : "sender"
   );
-  const txAddresses = transactions.map((t) =>
-    direction === Direction.IN ? t.sender.address : t.target.address
+
+  const recipients: Wallet[] = transactions.map((t) =>
```

Comment:
La variable contient les destinataires ou les emetteurs, selon la direction. 
```suggestion
  const txWallets: Wallet[] = transactions.map((t) =>
```